x/mongo/driver/operation/update.go

```diff
@@ -23,6 +24,7 @@ import (
 // Update performs an update operation.
 type Update struct {
 	bypassDocumentValidation *bool
+	comment                  bsoncore.Value
```

Comment:
Is there a reason to use a `bsoncore.Value` here over a `bsoncore.Document` like the `let` field?

Reply:
The comment can be any valid BSON type, according to the [specifications](https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst):

```
   * The comment can be any valid BSON type for server versions 4.4 and above.
   * Server versions prior to 4.4 do not support comment for insert command,
   * and providing one will result in a server-side error.
```

The spec tests that are given focus on strings and documents, in particular.  Given this, does `bsoncore.Value` seem like the correct type to be using here?  I may have a misunderstanding of what `bsoncore.Document` actually does.

Reply:
Ah, I see the note about "any valid BSON type" in the spec now. I also see that the spec mentions that `let` should specifically be a document:
```
let: Optional<Document>;
```

I previously thought that a document or a string were the only two expected values. Seems like `bsoncore.Value` is the most correct type to use for "any valid BSON type". Thanks for the explanation!